### PR TITLE
Agent: enable testing mode that panics on sync errors

### PR DESCRIPTION
**Pull request by olafurpg**
## Test plan
- Validated that on runIde the agent panics and logs on desync issues

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -369,9 +380,13 @@ tasks {
   runIde {
     dependsOn(project.tasks.getByPath("buildCody"))
     jvmArgs("-Djdk.module.illegalAccess.silent=true")
+    // TODO: Figure out why systemProperties(...) doesn't work
```
Comment (RXminuS) [inline build.gradle.kts]: It's because one was `cody-agent` and the other is `cody.agent`. Fix incoming...

Comment (olafurpg) [reply build.gradle.kts]: Wow, nice catch (and thanks Cody for suggesting the wrong key 😢 )

File: src/main/kotlin/com/sourcegraph/cody/agent/CodyAgent.kt [hunk 2]
```diff
@@ -287,7 +287,7 @@ private constructor(
     }
 
     private fun traceWriter(): PrintWriter? {
-      val tracePath = System.getProperty("cody-agent.trace-path", "")
+      val tracePath = System.getProperty("cody.agent.trace-path", "")
```
Comment (olafurpg) [inline src/main/kotlin/com/sourcegraph/cody/agent/CodyAgent.kt]: We should not change this as we have existing (internal dev) usage of this property (example: in my VM options).

Comment (RXminuS) [reply src/main/kotlin/com/sourcegraph/cody/agent/CodyAgent.kt]: Ok switched it to kebab-case again

**Review by RXminuS — APPROVED** [review]

**Review by pkukielka — APPROVED** [review]
LGTM

**Pull request merged**